rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+                    'X-BAPI-TIMESTAMP': str(timestamp),
+                    'X-BAPI-SIGN': signature,
+                }
+            else:
+                self.session.headers = {'Content-Type': 'application/json', 'X-BAPI-SIGN-TYPE': '2'}  # noqa: E501
+
+            try:
+                response = self.session.request(
+                    method='get',
+                    url=url,
+                    params=options,
+                    timeout=timeout,
+                )
+            except requests.exceptions.RequestException as e:
+                raise RemoteError(f'Bybit API request failed due to {e}') from e
+
+            if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+                if tries >= 1:
+                    backoff_seconds = 10 / tries
+                    log.debug(
+                        f'Got a 429 from Bybit. Backing off for {backoff_seconds}')
+                    gevent.sleep(backoff_seconds)
+                    tries -= 1
+                    continue
+
+                # else
+                raise RemoteError(
+                    f'Bybit api request for {response.url} failed with HTTP '
+                    f'status code {response.status_code} and response {response.text}',
+                )
+
+            if response.status_code != HTTPStatus.OK:
+                raise RemoteError(
+                    f'Bybit api request for {response.url} failed with HTTP status '
+                    f'code {response.status_code} and response {response.text}',
+                )
+
+            break  # else all good, we can break off the retry loop
+
+        log.debug(f'BYBIT RESPONSE: {response.text}')
+        try:
+            json_ret = json.loads(response.text)
+        except json.JSONDecodeError as e:
+            raise RemoteError('Bybit returned invalid JSON response') from e
+
+        if 'result' not in json_ret:
+            raise RemoteError(f'Remote response is missing expected field result: {json_ret}')
+
+        if json_ret['retCode'] != 0 and json_ret['retMsg'] != 'OK':
+            raise RemoteError(f'Bybit returned error in request {json_ret}')
+
+        return json_ret['result']
+
+    def _paginated_api_query(
+            self,
+            endpoint: Literal[
+                'order/history',
+                'asset/deposit/query-record',
+                'asset/withdraw/query-record',
+            ],
+            options: dict[str, Any],
+    ) -> list[dict[str, Any]]:
+        if endpoint == 'order/history':
+            result_key = 'list'
+        else:
+            result_key = 'rows'
+
+        result = []
+        while True:
+            output = self._api_query(path=endpoint, options=options)
+            if (next_cursor := output.get('nextPageCursor')) is not None:
+                options['cursor'] = next_cursor
+
+            result.extend(output[result_key])
+            log.debug(f'Got {len(output[result_key])} items in the query')
```

Comment:
This mentions nothing about bybit and not sure what's the point here is? Is it for your own debugging?